File: 4C.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int t;
    vector<string> usernames;
    int i = 0;
    while(t--){
        string name;
        cin >> name;
        int count = 0;
        for(int i = 0; i < usernames.size(); i++){
            if(usernames[i] == name){
                count++;
            }
        }
    }
    return 0;
}

File: 4C.py
t = int(input())
i = 0
usernames = dict()
while(i < t):
    name = input()
    if(name in usernames.keys()):
        print(name + str(usernames[name]))
        usernames[name] += 1
    else:
        usernames[name] = 1
        print("OK")
    i += 1